R/sql-clause.R

```diff
@@ -15,79 +16,124 @@ sql_select_clauses <- function(con,
     order_by = order_by,
     limit = limit
   )
-  out <- purrr::compact(out)
-  escape(unname(out), collapse = "\n", parens = FALSE, con = con)
+  sql_format_clauses(out, lvl, con)
 }
 
-sql_clause_select <- function(con, select, distinct = FALSE, top = NULL) {
+sql_clause <- function(kw, parts, sep = ",", parens = FALSE, lvl = 0) {
+  list(
```

Comment:
Is it worth giving this a class? If only to make debugging/exploration a little easier?